openapi/openapi_client.go

```diff
@@ -83,6 +86,9 @@ func (o *ProviderClient) performRequest(method httpMethodSupported, resourceURL
 	o.appendOperationHeaders(operation.HeaderParameters, o.providerConfiguration, reqContext.headers)
 	log.Printf("[DEBUG] Performing %s %s", method, reqContext.url)
 
+	userAgentHeader := version.BuildUserAgent(runtime.GOOS, runtime.GOARCH)
+	o.appendUserAgentHeader(reqContext.headers, userAgentHeader)
```

Comment:
agreed, the TestPerformRequest will need to be updated to capture this new behaviour..the HttpClientStub{} is prepared to capture the headers used in all the operations (post, put, remove, get), we can make use of that to assert that the expected headers are being sent

Reply:
I added a check in ```TestPerformRequest``` to make sure that the User-Agent key exists in the header map. However, I do not assert the value of the header, as it may or may not require monkey patching ```BuildUserAgent